include/mqtt/property.hpp

```diff
@@ -152,117 +148,26 @@ struct binary_property {
         return 2;
     }
 
-     std::string const& val() const {
+     buffer const& val() const {
         return buf_;
     }
 
     static constexpr ostream_format const of_ = ostream_format::direct;
     property::id id_;
+    buffer buf_;
     boost::container::static_vector<char, 2> length_;
```

Comment:
I'm not sure you saw this comment

Reply:
Sorry, I looked over it.
I think that `length_` is stored as static_vector because of `add_const_buffer_sequence()`.

See 

```cpp
    void add_const_buffer_sequence(std::vector<as::const_buffer>& v) const {
        v.emplace_back(as::buffer(&id_, 1));
        v.emplace_back(as::buffer(length_.data(), length_.size())); // as::buffer is view. We need to store body somewhare.
        v.emplace_back(as::buffer(buf_.data(), buf_.size()));
    }
```

Reply:
I really wish there was a way to do this without storing it indefinitely.

Not something that needs to be fixed for this PR, but maybe we can make improvements to this later.